File: 2425. Bitwise XOR of All Pairings/main.py
class Solution:
    def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
        # If the length of array is even, XOR all elements of arry is 0
        xor1 = 0
        xor2 = 0
        
        for num in nums1:
            xor1 ^= num

        for num in nums2:
            xor2 ^= num
        
        if len(nums2) % 2 == 0:
            xor1 = 0
        if len(nums1) % 2 == 0:
            xor2 = 0
        
        return xor1 ^ xor2

File: 2425. Bitwise XOR of All Pairings/main.java
class Solution {
    public int xorAllNums(int[] nums1, int[] nums2) {
        int xor1 = 0;
        int xor2 = 0;

        for(int i=0;i<nums1.length;i++){
            xor1 ^= nums1[i];
        }
    
        for(int i=0;i<nums2.length;i++){
            xor2 ^= nums2[i];
        }

        if(nums1.length%2==0){
            xor2 = 0;
        }
        if(nums2.length%2==0){
            xor1 = 0;
        }

        return xor1^xor2;
    }
}